libraries/USB/src/USBMIDI.cpp

```diff
@@ -41,9 +48,55 @@ USBMIDI::USBMIDI() {
   }
 }
 
+// private function for setting a not null/empty MIDI device name limited to 32 characters
+void USBMIDI::setDeviceName(const char* name) {
+    const uint8_t maxNameLength = 32; // tinyUSB Descriptor limit
+    if (name != nullptr && strlen(name) > 0) {
+      if (strlen(name) > maxNameLength) {
+        log_w("USBMIDI: Device name too long, truncating to %d characters.", maxNameLength);
+      }
+      midiUserDeviceName = new char[maxNameLength + 1]; // +1 for null-terminator
+      if (midiUserDeviceName) {
+        strncpy(midiUserDeviceName, name, maxNameLength);
+        // Ensure null-termination when overflowing
+        midiUserDeviceName[maxNameLength] = '\0';
+      } else {
+        log_e("USBMIDI: Failed to allocate memory for device name, using default name.");
+      }
+    } else {
+      log_w("USBMIDI: No device name provided, using default name [%s].", getUSBMIDIDefaultDeviceName());
+    }
+}
+
+USBMIDI::USBMIDI(const char* name) {
```

Comment:
```suggestion
/**
* @brief Get the current device name
* @return The device name in order of precedence:
* 1. Name set via constructor (if any)
* 2. Name set via SET_USB_MIDI_DEVICE_NAME() macro (if defined)
* 3. Default name "TinyUSB MIDI"
* If device name is set as "", it will be ignored
*/
USBMIDI::USBMIDI(const char* name) {
```